core/tauri/src/manager.rs

```diff
@@ -1005,23 +1007,46 @@ impl<R: Runtime> WindowManager<R> {
       ));
     }
 
-    #[cfg(feature = "window-data-url")]
-    if let Some(csp) = self.csp() {
-      if url.scheme() == "data" {
-        if let Ok(data_url) = data_url::DataUrl::process(url.as_str()) {
-          let (body, _) = data_url.decode_to_vec().unwrap();
-          let html = String::from_utf8_lossy(&body).into_owned();
-          // naive way to check if it's an html
-          if html.contains('<') && html.contains('>') {
+    match (
+      url.scheme(),
+      tauri_utils::html::extract_html_content(url.as_str()),
+    ) {
+      #[cfg(feature = "window-data-url")]
+      ("data", Some(html_string)) => {
+        // There is an issue with the external DataUrl where HTML containing special characters
+        // are not correctly processed. A workaround is to first percent encode the html string,
+        // before it processed by DataUrl.
+        let encoded_string = percent_encoding::utf8_percent_encode(html_string, percent_encoding::NON_ALPHANUMERIC).to_string();
```

Comment:
we shouldn't encode the URL, we should expect the user has already encoded it as per https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs 
which makes this PR kinda uneeded?

Reply:
I expect tauri not to encode URLs by default.

The following content doesn't work.
```js
const webview = new WebviewWindow('print', {
        url: `data:text/html,<html><body>你好，世界</body></html>`,
        center: true,
        visible: true,
        width: 300,
        height: 300,
});

```